tests/v2/validation/Jenkinsfile_pre_post_upgrade

```diff
@@ -0,0 +1,167 @@
+                    def decoded = new String(AWS_SSH_PEM_KEY.decodeBase64())
+                    writeFile file: AWS_SSH_KEY_NAME, text: decoded
+                  }
+                }
+
+                dir("./tests/v2/validation") {
+                  def filename = "config.yaml"
+                  def configContents = env.CONFIG
+
+                  writeFile file: filename, text: configContents
+                  env.CATTLE_TEST_CONFIG = "/go/src/github.com/rancher/rancher/tests/v2/validation/"+filename
+                }
+
+                sh "./tests/v2/validation/configure.sh"
+                sh "./tests/v2/validation/build.sh"
+              }
+              stage('Run Validation Tests') {
+                try {
+                  echo envFile
+                  sh "docker run --name ${testContainer} -t --env-file ${envFile} " +
+                  "${imageName} sh -c \"/go/bin/gotestsum --format standard-verbose --packages=${testsDir} --junitfile ${testResultsOut} -- ${GOTEST_TESTCASE} -timeout=${timeout} -v\""
+                } catch(err) {
+                  echo 'Test run had failures. Collecting results...'
+                }
+              }
+              stage('upgrade') {
+                  upgradeParams = [
+                  string(name: 'RANCHER_HA_HOSTNAME', value: "${RANCHER_HA_HOSTNAME}"),
+                  string(name: 'RANCHER_CHART_VERSION', value: "${RANCHER_CHART_VERSION}"),
+                  string(name: 'RANCHER_HA_CERT_OPTION', value: "${RANCHER_HA_CERT_OPTION}"),
+                  string(name: 'RANCHER_HA_KUBECONFIG', value: "${RANCHER_HA_KUBECONFIG}")
+                  ]
+                  build job: 'rancher-v3_ha_upgrade', parameters: upgradeParams
+              }
+                            
+              stage ('post upgrade tests') {
+                try {
+                  sh "docker run --name ${upgradeTestContainer} -t --env-file ${envFile} " +
+                    "${imageName} sh -c \"/go/bin/gotestsum --format standard-verbose --packages=${testsDir} --junitfile ${testResultsOut} -- ${GOTEST_TESTCASE_2} -timeout=${timeout} -v\""
+                } catch(err) {
+                  echo 'Test run had failures. Collecting results...'
+                }
+              }
+              stage('delete') {
+                CATTLE_TEST_URL = "http://${RANCHER_HA_HOSTNAME}"
+                deleteParams = [
+                string(name: 'CATTLE_TEST_URL', value: "${CATTLE_TEST_URL}"),
+
+                ]
+                build job: 'rancher-v3_ha_delete', parameters: deleteParams
+              }
+                     
+              stage('Test Report') {
+                sh "docker cp ${testContainer}:${rootPath}${testResultsOut} ."
+                step([$class: 'JUnitResultArchiver', testResults: "**/${testResultsOut}"])
+                sh "docker stop ${testContainer}"
+                sh "docker rm -v ${testContainer}"
+                sh "docker rmi -f ${imageName}"
```

Comment:
I think we need to stop and remove `upgradeTestContainer` here like `testContainer`.

Reply:
added stop and remove for  `upgradeTestContainer`